### DISCUSSION TITLE
Unavailable to get my bets history

### DISCUSSION THREAD

**Discussion by vy-developer** [cloudbet/docs#31] [Q&A]
Trying to get my bets history though API, but every time when i make a request to Server i get response with 0 bets. 

`{
  "bets": [],
  "totalBets": "0"
}
`
I try to use Cloudbet API page to test a request but still nothing and also i tried to use this simple script on python but same results. 

````
import requests

session = requests.Session()
session.headers["X-API-Key"] = ""

url = "https://sports-api.cloudbet.com/pub/v3/bets/history"

offset = 0
limit = 20
bets = []
while True:
    response_data = session.get(url, params={"limit": limit, "offset": offset}).json()
    bets += response_data["bets"]
    if len(response_data["bets"]) < limit:
        break
    offset += 20

print(f"{'EVENT':52} {'MARKET':50} {'STAKE':6} {'PRICE':7} {'STATUS':9} {'PROFIT'}")
for bet in bets:
    print(
        "{eventName:52} {marketUrl:50} {stake:6} {price:7} {status:9} {returnAmount}"
        .format_map(bet)
    )
````

**Comment by kgravenreuth** [comment]
Hi @NoNam44 

The public bet history endpoint displays the bets which you have placed through the API. It does not show the bets you might have placed via the cloudbet.com website manually. So please place bets via the Cloudbet API and try again.

We'll improve our [Cloudbet Betting API Docs](https://www.cloudbet.com/api/?utm_source=github.com&utm_medium=social&utm_campaign=axis&urls.primaryName=Trading#/Trading/BetHistoryV4) to clarify this.

Let me know if you have any other questions.

Cheers,
Klaus Gravenreuth
Product & Engineering @ Cloudbet
